External/dav1d/CMakeLists.txt

```diff
@@ -0,0 +1,368 @@
+include(External)
+
+# git clone -b 1.5.0 https://code.videolan.org/videolan/dav1d.git
+# in llvm-test-suite/test-suite-externals.
+
+llvm_externals_find(TEST_SUITE_DAV1D_ROOT "dav1d" "dav1d 1.5.0")
+
+if (NOT TEST_SUITE_DAV1D_ROOT)
+  return()
+endif()
+
+include(CheckCCompilerFlag)
+include(CheckFunctionExists)
+include(CheckLanguage)
+include(CheckLibraryExists)
+include(CheckLinkerFlag)
+
+set(CMAKE_C_STANDARD 17)
+
+include_directories(.)
+include_directories(${TEST_SUITE_DAV1D_ROOT}/include)
+include_directories(${TEST_SUITE_DAV1D_ROOT}/include/dav1d)
+include_directories(${TEST_SUITE_DAV1D_ROOT})
+include_directories(${TEST_SUITE_DAV1D_ROOT}/src)
+
+if (WIN32)
+  include_directories(${TEST_SUITE_DAV1D_ROOT}/include/compat)
+endif()
+
+function(check_enable_option option)
+  string(REGEX REPLACE "^--*" "" option_name ${option})
+  string(TOUPPER ${option_name} option_name)
+  string(REGEX REPLACE "[-=]" "_" option_name ${option_name})
+  set(varname "SUPPORTS_${option_name}")
+  check_c_compiler_flag(${option} ${varname})
+  if (${varname})
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${option}" PARENT_SCOPE)
+  endif()
+endfunction()
+
+function(check_disable_warning warning)
+  string(REGEX REPLACE "^-Wno-" "" warning ${warning})
+  string(TOUPPER ${warning} warning_name)
+  string(REGEX REPLACE "[-=]" "_" warning_name ${warning_name})
+  set(varname "SUPPORTS_W${warning_name}")
+  check_c_compiler_flag(-W${warning} ${varname})
+  if (${varname})
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-${warning}" PARENT_SCOPE)
+  endif()
+endfunction()
+
+if (NOT MSVC)
+  # clang-cl supports -Wall, but it corresponds to -Weverything
+  check_enable_option(-Wall)
+endif()
+
+check_enable_option(-Wundef)
+check_enable_option(-Werror=vla)
+check_disable_warning(-Wno-maybe-uninitialized)
+check_disable_warning(-Wno-missing-field-initializers)
+check_disable_warning(-Wno-unused-parameter)
+check_enable_option(-Wstrict-prototypes)
+check_enable_option(-Werror=missing-prototypes)
+check_enable_option(-Wshorten-64-to-32)
+
+check_function_exists(sin HAVE_DEFAULT_MATH)
+if (NOT HAVE_DEFAULT_MATH)
+  check_library_exists(m sin "" HAVE_LIBM)
+  if (HAVE_LIBM)
+    link_libraries(m)
+  endif()
+endif()
+check_library_exists(atomic __atomic_load_8 "" HAVE_LIBATOMIC)
+if (HAVE_LIBATOMIC)
+  link_libraries(atomic)
+endif()
+if (NOT WIN32)
+  find_package(Threads)
+  if (Threads_FOUND)
+    link_libraries(${CMAKE_THREAD_LIBS_INIT})
+  endif()
+endif()
+
+if (WIN32)
+  add_compile_definitions(WIN32_LEAN_AND_MEAN)
+  if (MSVC)
+    add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
+    add_compile_definitions(_CRT_NONSTDC_NO_DEPRECATE)
+  endif()
+endif()
+
+if (CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+  enable_language(ASM)
+  set(CMAKE_SYSTEM_PROCESSOR "aarch64")
```

Comment:
Why do we need to adjust this? If it is just needed locally in the file, better use a new variable instead of (re-)setting an existing one.

Reply:
Yeah, it's only used locally within the file.

As CMake isn't very consistent about `CMAKE_SYSTEM_PROCESSOR` (you'd have different values for aarch64/arm64 on different OSes, likewise for x86), I tried to normalize it here (to a value which is a potential value for `CMAKE_SYSTEM_PROCESSOR` anyway), to simplify the later checks.

Originally I had prototyped the CMake wrapping with CMakeLists.txt within the dav1d (and ffmpeg) source tree, with one CMakeLists.txt per subdirectory, so there it felt more natural to just use the standard variable, but assuming normalization of it. But when we have it all within one single file, I guess it's just as easy and clear to just use new variables for it as well.